color/src/parse.rs

```diff
@@ -482,43 +485,70 @@ pub fn parse_color(s: &str) -> Result<DynamicColor, ParseError> {
                     Err(ParseError::UnknownColorIdentifier)
                 }
             }
-        }
-        // TODO: should we validate that the parser is at eof?
+        }?;
+        Ok((parser.ix, color))
     } else {
         Err(ParseError::UnknownColorSyntax)
     }
 }
 
-const fn get_4bit_hex_channels(hex_str: &str) -> Result<[u8; 8], ParseError> {
-    let mut four_bit_channels = match *hex_str.as_bytes() {
-        [r, g, b] => [r, r, g, g, b, b, b'f', b'f'],
-        [r, g, b, a] => [r, r, g, g, b, b, a, a],
-        [r0, r1, g0, g1, b0, b1] => [r0, r1, g0, g1, b0, b1, b'f', b'f'],
-        [r0, r1, g0, g1, b0, b1, a0, a1] => [r0, r1, g0, g1, b0, b1, a0, a1],
-        _ => return Err(ParseError::WrongNumberOfHexDigits),
-    };
+// Arguably this should be an implementation of FromStr.
+/// Parse a color string in CSS syntax into a color.
+///
+/// This parses the entire string; trailing characters cause an
+/// [`ExpectedEndOfString`](ParseError::ExpectedEndOfString) parse error. See also
+/// [`parse_color_with_remainder`].
+///
+/// # Errors
+///
+/// Tries to return a suitable error for any invalid string, but may be
+/// a little lax on some details.
+pub fn parse_color(s: &str) -> Result<DynamicColor, ParseError> {
+    let (ix, color) = parse_color_with_remainder(s)?;
+
```

Comment:
I think it would be nice to skip trailing whitespace.

Of course, there's the related question of whether we should also skip leading whitespace. Currently we don't, but that would be easy enough.